Ponca/src/Fitting/mean.h

```diff
@@ -92,12 +94,35 @@ namespace Ponca {
         PONCA_FITTING_DECLARE_INIT
         PONCA_FITTING_DECLARE_ADDNEIGHBOR_DER
 
-        /// \brief Compute derivatives of the barycenter. \see MeanPosition::barycenter()
+        /// \brief Compute derivatives of the barycenter.
+        /// \see MeanPosition::barycenter()
+        ///
+        /// ### Step-by-step derivation from the barycenter definition
+        /// Given the definition of the barycenter \f$ b(\mathbf{x}) = \frac{\sum w(\mathbf{x}) \mathbf{x}}{\sum w(\mathbf{x})} \f$ :
+        ///
+        /// We denote \f$ t(\mathbf{x}) = \sum w(\mathbf{x}) \mathbf{x} \f$ and \f$ s(\mathbf{x}) = \sum w(\mathbf{x})\f$,
+        /// such that \f$ b(\mathbf{x}) = \frac{t(\mathbf{x})}{s(\mathbf{x})}\f$.
+        ///
+        /// By definition, \f$ b'(\mathbf{x}) = \frac{s(\mathbf{x})t'(\mathbf{x}) - t(\mathbf{x})s'(\mathbf{x})}{s(\mathbf{x})^2}\f$.
+        /// We have \f$ s'(\mathbf{x}) = \sum w'(\mathbf{x}) \f$.
+        ///
+        /// We rewrite \f$ t(\mathbf{x}) = \sum u(\mathbf{x})v(\mathbf{x}) \f$, with \f$ u(\mathbf{x}) = w(\mathbf{x}) \f$ and \f$ v(\mathbf{x}) = \mathbf{x} \f$.
+        ///
+        /// By definition, \f$ t'(\mathbf{x}) = \sum u(\mathbf{x})v'(\mathbf{x}) + u'(\mathbf{x})v(\mathbf{x}) \f$, with  \f$ u'(\mathbf{x}) = w'(\mathbf{x})\f$ and \f$ v'(\mathbf{x}) = \mathbf{1}\f$.
+        ///
+        /// Which leads to \f$ b'(\mathbf{x}) = \frac{\sum w(\mathbf{x})\left(\mathbf{1} \sum w(\mathbf{x}) + w'(\mathbf{x})\mathbf{x}\right) - \left(\sum w(\mathbf{x})\mathbf{x}\right)\sum w'(\mathbf{x})}{\left(\sum w(\mathbf{x})\right)^2} \f$
+        ///
+        /// Simplifying by \f$ \sum w(\mathbf{x}) \f$ we obtain:
+        ///
+        /// \f$ b'(\mathbf{x}) = \frac{\mathbf{1} \sum w(\mathbf{x})  + w'(\mathbf{x})\mathbf{x} - b(\mathbf{x})\sum w'(\mathbf{x})}{\sum w(\mathbf{x})} \f$
```

Comment:
I think there is still an error because the derivative of a point is null, so `v' = 0` instead of `v' = 1` (the points denoted here by `x` are actually constant points `x_i` from the point cloud, and not variable with respect to scale nor space).
So `t' = \sum u' v = \sum w' x`, and then `b' = ( \sum w' x - b \sum w' ) / \sum w`. 

I believe this can be compactly computed in one line by `VectorArray barycenterDer = ( m_dSumP - Base::barycenter() * Base::m_dSumW ) / Base::m_sumW`.

Reply:
Thanks for the review, I updated the code and the documentation accordingly, emphasizing the difference between x and p_i.